ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -3128,17 +3129,27 @@ void terminate_registered_assignment(ast_node_t* always_node, signal_list_t* ass
         }
     }
 
+#warning "Oops this be broke come fix"
     for (i = 0; i < assignment->count; i++) {
         npin_t* pin = assignment->pins[i];
-        dependence_variable_position = -1;
+        int dependence_variable_position = -1;
 
-        if (pin->net->driver_pin) {
-            ref_string = pin->net->driver_pin->node->name;
+        if(pin->net->num_driver_pins > 1) {
+            // TODO Come make this warning actually correct when I know what is happening
+            error_message( NETLIST, pin->node->line_number, pin->node->file_number, "Multiple drivers not supported for assignment to %s", pin->name);
+        }
+        if (pin->net->num_driver_pins) {
+            ref_string = pin->net->driver_pins[0]->node->name;
 
             for (j = i - 1; j >= 0; j--) {
-                if (list_dependence_pin[j] && list_dependence_pin[j]->net->driver_pin && list_dependence_type[j] && list_dependence_type[j] == BLOCKING_STATEMENT && strcmp(ref_string, assignment->pins[j]->node->name) == 0) {
+                if(list_dependence_pin[j] && list_dependence_pin[j]->net->num_driver_pins > 1)
+                {
+                    // TODO Come make this warning actually correct when I know what is happening
```

Comment:
do you have an example bit of Verilog code that can make this reachable? I'm just wondering if you could not just attach all the drivers anyways

Reply:
Im not sure how do go about doing that. Hence the obnoxious warning message. I will look into it some more.

Reply:
I tried a few things to hit this, but I cant seem to find a case that has a non-null `assignment->pins[j]->node`. Any ideas?

```
module scratch(input clk, input in, output out);
reg a;
reg b;
always @(posedge clk)
begin
    a = in;
    b = a;
end
endmodule
```

Enters that loop. But I cant get it to actually do anything